pkg/repository/prisma/dbsqlc/events.sql

```diff
@@ -147,27 +150,77 @@ SELECT
 FROM
     "Event" as events
 WHERE
+    events."deletedAt" IS NOT NULL AND
     "tenantId" = @tenantId::uuid AND
     "id" = ANY (sqlc.arg('ids')::uuid[]);
 
--- name: DeleteExpiredEvents :one
-WITH expired_events_count AS (
-    SELECT COUNT(*) as count
+-- name: SoftDeleteExpiredEvents :one
+WITH has_more AS (
```

Comment:
I think we are still risking having the count returned within `has_more` be different from the count within `expired_events_with_limit` since we're skipping locked in 1 case but not the other. 

Perhaps place the lock on all rows (the `+ 1` row included), and just use a row_number to filter by `<= limit` from the CTE later on?